### PR TITLE
I3S-attribute-loader - Fix for attributes loading

### PR DESCRIPTION
**Pull request by dryabinin-actionengine**
This fix allows to load all attributes and doesn't fail all loaded attributes if any of them are not fetched or parsed properly

### CODE REVIEW

File: modules/i3s/src/helpers/promises-handler.js [hunk 1]
```diff
@@ -0,0 +1,22 @@
+export const REJECTED_STATUS = 'rejected';
+export const FULFILLED_STATUS = 'fulfilled';
+
+/**
+ * Handle list of promises based on browser compatibility
+ * @param {Promise[]} promises
+ * @returns {Promise}
+ */
+export function handlePromises(promises) {
+  return Promise.allSettled ? Promise.allSettled(promises) : Promise.all(promises.map(reflect));
```
Comment (ibgreen) [inline modules/i3s/src/helpers/promises-handler.js:10]: Which browser is having issues with allSettled? Maybe mention in comment?

We could also just polyfill `allSettled`?

File: modules/i3s/src/i3s-attribute-loader.js [hunk 2]
```diff
@@ -55,7 +55,7 @@ export async function loadFeatureAttributes(tile, featureId, options = {}) {
     attributeLoadPromises.push(promise);
   }
   try {
-    attributes = await Promise.all(attributeLoadPromises);
+    attributes = await handlePromises(attributeLoadPromises);
```
Comment (ibgreen) [inline modules/i3s/src/i3s-attribute-loader.js:58]: Maybe add a comment explaining why we need to support failed promises?

**Review by ibgreen — APPROVED** [review]

File: modules/i3s/src/i3s-attribute-loader.js (hunk 2 above)
Comment (dryabinin-actionengine) [reply modules/i3s/src/i3s-attribute-loader.js:58]: Ok. Actually some of attributes can be encoded badly or some attributes can't to be loaded. We should get as many attributes as possible to show available attributes for user.

File: modules/i3s/src/helpers/promises-handler.js (hunk 1 above)
Comment (dryabinin-actionengine) [reply modules/i3s/src/helpers/promises-handler.js:10]: `Promise.allSettled() `was implemented in `ECMAScript 2020, the 11th edition`. It has wide browser support but pretty much for newest versions of browsers:

image

For example `Promise.all() ` support table:
image

Yes I think we can add a polyfill for it.

**Pull request merged**